QRCoder/PayloadGenerator/Girocode.cs

```diff
@@ -36,16 +36,16 @@ public class Girocode : Payload
         /// <param name="version">Girocode version. Either 001 or 002. Default: 001.</param>
         /// <param name="encoding">Encoding of the Girocode payload. Default: ISO-8859-1</param>
         /// <exception cref="GirocodeException">Thrown when the input values are not valid according to the Girocode specification.</exception>
-        public Girocode(string iban, string bic, string name, decimal amount, string remittanceInformation = "", TypeOfRemittance typeOfRemittance = TypeOfRemittance.Unstructured, string purposeOfCreditTransfer = "", string messageToGirocodeUser = "", GirocodeVersion version = GirocodeVersion.Version1, GirocodeEncoding encoding = GirocodeEncoding.ISO_8859_1)
+        public Girocode(string iban, string? bic, string name, decimal amount, string remittanceInformation = "", TypeOfRemittance typeOfRemittance = TypeOfRemittance.Unstructured, string purposeOfCreditTransfer = "", string messageToGirocodeUser = "", GirocodeVersion version = GirocodeVersion.Version1, GirocodeEncoding encoding = GirocodeEncoding.ISO_8859_1)
         {
             _version = version;
             _encoding = encoding;
             if (!IsValidIban(iban))
                 throw new GirocodeException("The IBAN entered isn't valid.");
             _iban = iban.Replace(" ", "").ToUpper();
-            if (!IsValidBic(bic))
+            if (!IsValidBic(bic, _version == GirocodeVersion.Version1))
                 throw new GirocodeException("The BIC entered isn't valid.");
-            _bic = bic.Replace(" ", "").ToUpper();
+            _bic = bic?.Replace(" ", "").ToUpper() ?? string.Empty;
```

Comment:
```suggestion
            _bic = bic?.Replace(" ", "").ToUpper() ?? "";
```
To a huge part a matter of taste, but using the literal `""` has a few advantages over using `string.Empty` (in general, not specifi to this repo):
* less IL code as the literal is a constant as opposed a field load -> smaller assembly
* less work for the JIT, as no load and inlining of that field is needed

If a change is done, we should it do for the whole project to be consistent.
If not, I'm also fine with it, as these two listed bullet don't change much here.

This is more a general note.

Reply:
In general, I'd like to follow .NET analysis rules. I'm not sure exactly what that means in this specific case, but one of my goals is to enable the analyzers at the recommended level. It might not be practical until we remove support for .NET Framework in v2...

Reply:
#638